localstack/services/stepfunctions/provider.py

```diff
@@ -111,10 +112,31 @@ def get_store(context: RequestContext) -> SFNStore:
     def accept_state_visitor(self, visitor: StateVisitor):
         visitor.visit(sfn_stores)
 
+    _STATE_MACHINE_ARN_REGEX: Final[re.Pattern] = re.compile(
+        r"^arn:aws:states:[a-z0-9-]+:[0-9]{12}:stateMachine:[a-zA-Z0-9-_]+(:\d+)?$"
+    )
+
+    _STATE_MACHINE_EXECUTION_ARN_REGEX: Final[re.Pattern] = re.compile(
+        r"^arn:aws:states:[a-z0-9-]+:[0-9]{12}:(stateMachine|execution):[a-zA-Z0-9-_]+(:\d+)?(:[a-zA-Z0-9-_]+)?$"
+    )
+
+    @staticmethod
+    def _validate_state_machine_arn(state_machine_arn: str) -> None:
+        if not StepFunctionsProvider._STATE_MACHINE_ARN_REGEX.match(state_machine_arn):
+            raise InvalidArn(f"Invalid arn: '{state_machine_arn}'")
+
+    @staticmethod
+    def _raise_state_machine_does_not_exist(state_machine_arn: str) -> None:
+        raise StateMachineDoesNotExist(f"State Machine Does Not Exist: '{state_machine_arn}'")
+
+    def _validate_state_machine_execution_arn(self, execution_arn: str) -> None:
+        if not StepFunctionsProvider._STATE_MACHINE_EXECUTION_ARN_REGEX.match(execution_arn):
+            raise InvalidArn(f"Invalid arn: '{execution_arn}'")
```

Comment:
Please add a comment to document the lack of parity here with the observed AWS behavior (e.g. the error messages complaining about the ARN prefix). Just a general remark so that we don't forget about this disparity is enough :+1: